### PR TITLE
[Draft]: Stable Diffusion quantization example

### PR DESCRIPTION
**Pull request by AlexKoff88**

### CODE REVIEW

File: examples/openvino/stable-diffusion-quantization/quantize.py [hunk 1]
```diff
@@ -0,0 +1,1030 @@
...
+        unet = export_unet,
+        tokenizer=tokenizer,
+        scheduler=noise_scheduler,
+        safety_checker=pipeline.safety_checker,
+        feature_extractor=pipeline.feature_extractor,
+    )
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        export_to_onnx(export_pipeline, tmpdirname)
+        export_to_openvino(export_pipeline, tmpdirname, Path(args.output_dir) / "openvino")
```
Comment (echarlaix) [inline examples/openvino/stable-diffusion-quantization/quantize.py:1024]: It would be nice to have an export function in optimum directly for this kind of case, will open a PR for this

**Review by echarlaix — COMMENTED** [review]
It looks great @AlexKoff88 ! 🔥 Just left a couple of minor comments

File: examples/openvino/stable-diffusion-quantization/README.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+## HW Requirements for QAT
+The minimal HW setup for the run is GPU with 24GB of memory.
+
+>**NOTE**: Potentially you can set the number of training steps to 0 and it will lead to Post-Training Quantization. CPU should be enough in this case but you may need to modify the scipt.
+
+## Run QAT:
+* QAT for pokemon generation model:
+```python
+python quantize.py --ema_device="cpu" --use_kd --model_id="svjack/Stable-Diffusion-Pokemon-en" --center_crop --random_flip --gradient_checkpointing --dataloader_num_workers=2 --dataset_name="lambdalabs/pokemon-blip-captions"  --max_train_steps=4096 --opt_init_steps=300 --output_dir=sd-quantized-pokemon
+```
```
[the remark this answers is not in the document]
Comment (AlexKoff88) [reply examples/openvino/stable-diffusion-quantization/README.md]: Done

File: examples/openvino/stable-diffusion-quantization/README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+>**Note**: laion2B-en is being downloaded on-fly durint the fine-tuning process. No need to store it locally.
+
+## Prerequisites
+* Install Opimum-Intel for OpenVINO:
+```python
+pip install optimum-intel[openvino]
+```
+* Install example requirements:
+```python
+pip install -r requirements.txt
```
[the remark this answers is not in the document]
Comment (AlexKoff88) [reply examples/openvino/stable-diffusion-quantization/README.md:20]: Done

File: examples/openvino/stable-diffusion-quantization/README.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+The minimal HW setup for the run is GPU with 24GB of memory.
+
+>**NOTE**: Potentially you can set the number of training steps to 0 and it will lead to Post-Training Quantization. CPU should be enough in this case but you may need to modify the scipt.
+
+## Run QAT:
+* QAT for pokemon generation model:
+```python
+python quantize.py --ema_device="cpu" --use_kd --model_id="svjack/Stable-Diffusion-Pokemon-en" --center_crop --random_flip --gradient_checkpointing --dataloader_num_workers=2 --dataset_name="lambdalabs/pokemon-blip-captions"  --max_train_steps=4096 --opt_init_steps=300 --output_dir=sd-quantized-pokemon
+```
+
```
[the remark this answers is not in the document]
Comment (AlexKoff88) [reply examples/openvino/stable-diffusion-quantization/README.md:72]: I can do renaming, my idea was that the folder itself is called `stable-diffusion-quantization`.

[the remark this answers is not in the document]
Comment (AlexKoff88) [reply examples/openvino/stable-diffusion-quantization/README.md:72]: according to the Diffusers [layout](https://github.com/huggingface/diffusers/blob/main/examples/text_to_image), this script should be called like `train_text_to_image_qat.py`. @echarlaix what do you think?

File: examples/openvino/stable-diffusion-quantization/pokemon_generation_demo.ipynb [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    "This pipeline was fine-tuned on the public [dataset](https://huggingface.co/datasets/lambdalabs/pokemon-blip-captions) with Pokemon images and the correspoinding captions. You can find the model and the description [here](https://huggingface.co/svjack/Stable-Diffusion-Pokemon-en)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pipe = OVStableDiffusionPipeline.from_pretrained(\"svjack/Stable-Diffusion-Pokemon-en\", export=True, compile=False)\n",
```
[the remark this answers is not in the document]
Comment (AlexKoff88) [reply examples/openvino/stable-diffusion-quantization/pokemon_generation_demo.ipynb]: Uploaded the exported model to the group on the Hub.

File: notebooks/openvino/README.md [hunk 6]
```diff
@@ -12,4 +12,5 @@ The notebooks have been tested with Python 3.8 and 3.10 on Ubuntu Linux.
...
+| [Compare performance of the floating-point and quantized Stable Diffusion pipiline](https://github.com/huggingface/optimum-intel/blob/main/notebooks/openvino/question_answering_quantization.ipynb) | Show how to apply post-training quantization on a question answering model using [NNCF](https://github.com/openvinotoolkit/nncf) and to accelerate inference with OpenVINO| [![Open in Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/huggingface/optimum-intel/blob/main/notebooks/openvino/stable_diffusion_quantization.ipynb)| [![Open in AWS Studio](https://studiolab.sagemaker.aws/studiolab.svg)](https://studiolab.sagemaker.aws/import/github/huggingface/optimum-intel/blob/main/notebooks/openvino/stable_diffusion_quantization.ipynb)|
```
Comment (echarlaix) [inline notebooks/openvino/README.md:15]:
```suggestion
| [Compare outputs of a quantized Stable Diffusion model with its full-precision counterpart](https://github.com/huggingface/optimum-intel/blob/main/notebooks/openvino/stable_diffusion_quantization.ipynb) | Show how to load and compare outputs from two Stable Diffusion models with different precision| [![Open in Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/huggingface/optimum-intel/blob/main/notebooks/openvino/stable_diffusion_quantization.ipynb)| [![Open in AWS Studio](https://studiolab.sagemaker.aws/studiolab.svg)](https://studiolab.sagemaker.aws/import/github/huggingface/optimum-intel/blob/main/notebooks/openvino/stable_diffusion_quantization.ipynb)|
```

File: examples/openvino/stable-diffusion-quantization/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (echarlaix) [reply examples/openvino/stable-diffusion-quantization/README.md:72]: Actually in order to keep consistency we could rename it to `train_text_to_image.py`, and to leave the possibility to enable any NNCF supported compression techniques (potentially pruning in the future), while focusing on qat for the command line examples in the README. No strong opinion on this, `train_text_to_image_qat.py` is also working for me. Concerning the folder, I think `stable-diffusion` or `text-to-image` might be more adapted though

File: examples/openvino/stable-diffusion-quantization/pokemon_generation_demo.ipynb [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Let's fix the seed for reproducibility.\n",
+    "np.random.seed(42)\n",
+    "random.seed(42)\n",
+    "torch.manual_seed(42)\n",
```
[the remark this answers is not in the document]
Comment (echarlaix) [reply examples/openvino/stable-diffusion-quantization/pokemon_generation_demo.ipynb]: Yes and we could even use [`set_seed`](https://github.com/huggingface/diffusers/blob/v0.15.1/src/diffusers/training_utils.py#L32)  from `diffusers` directly

File: examples/openvino/stable-diffusion-quantization/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (AlexKoff88) [reply examples/openvino/stable-diffusion-quantization/README.md:72]: no prob, I'll do renaming.

File: examples/openvino/stable-diffusion-quantization/pokemon_generation_demo.ipynb (hunk 7 above)
[the remark this answers is not in the document]
Comment (AlexKoff88) [reply examples/openvino/stable-diffusion-quantization/pokemon_generation_demo.ipynb]: Done

File: examples/openvino/stable-diffusion-quantization/pokemon_generation_demo.ipynb [hunk 8]
```diff
@@ -0,0 +1,134 @@
+{
```
[the remark this answers is not in the document]
Comment (AlexKoff88) [reply examples/openvino/stable-diffusion-quantization/pokemon_generation_demo.ipynb]: moved

**Review by echarlaix — APPROVED** [review]

File: notebooks/openvino/stable_diffusion_quantization.ipynb [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "INFO:nncf:NNCF initialized successfully. Supported frameworks detected: torch, onnx, openvino\n"
+     ]
+    },
+    {
+     "ename": "ImportError",
```
[the remark this answers is not in the document]
Comment (AlexKoff88) [reply notebooks/openvino/stable_diffusion_quantization.ipynb:32]: Should I remove outputs at all?

[the remark this answers is not in the document]
Comment (helena-intel) [reply notebooks/openvino/stable_diffusion_quantization.ipynb:32]: I like outputs, including the generated images, so we can also link to it as an example of the kind of outputs users can expect.

**Review by helena-intel — APPROVED** [review]
